src/main/java/oshi/software/os/windows/WindowsFileSystem.java

```diff
@@ -52,53 +49,44 @@ public WindowsFileSystem() {
      */
     public OSFileStore[] getFileStores() {
-        // File.listRoots() has more information for Windows
-        // than FileSystem.getDefalut().getFileStores()
-        final File[] roots = File.listRoots();
-        // Need to call FileSystemView on Swing's Event Dispatch Thread to avoid
-        // problems
-        SwingWorker<List<OSFileStore>, Void> worker = new SwingWorker<List<OSFileStore>, Void>() {
-            @Override
-            public List<OSFileStore> doInBackground() {
-                FileSystemView fsv = FileSystemView.getFileSystemView();
-                List<OSFileStore> fsList = new ArrayList<>();
-                for (File f : roots) {
-                    String type = "unknown";
-                    String path = "unknown";
-                    try {
-                        // add trailing slash to path if needed
-                        path = f.getCanonicalPath();
-                        if (path.charAt(path.length() - 1) != '\\') {
-                            path = path + '\\';
-                        }
-                        char[] fstype = new char[16];
-                        if (Kernel32.INSTANCE.GetVolumeInformation(path, null, 0, null, null, null, fstype, 16)) {
-                            type = new String(fstype).trim();
-                        }
-                    } catch (IOException e) {
-                        LOG.error("Could not get canonical path for {}", f.toString());
-                    }
-                    fsList.add(new OSFileStore(fsv.getSystemDisplayName(f), path, fsv.getSystemTypeDescription(f), type,
-                            f.getUsableSpace(), f.getTotalSpace()));
-                }
-                return fsList;
-            }
-        };
-        worker.execute();
         List<OSFileStore> fs = new ArrayList<>();
-        try {
-            // TODO: Consider a timeout version of this method that passes
-            // timeout parameters which are used in this get()
-            fs = worker.get();
-        } catch (InterruptedException | ExecutionException e) {
-            LOG.error("", e);
+        WinDef.DWORD LENGTH = new WinDef.DWORD(255L);
+        char[] volume = new char[LENGTH.intValue()];
+
+        WinNT.HANDLE hVol = Kernel32.INSTANCE.FindFirstVolume(volume, LENGTH.intValue());
+        if (hVol == WinNT.INVALID_HANDLE_VALUE) {
+            return fs.toArray(new OSFileStore[0]);
+        }
+
+        while (true) {
+            char[] fstype = new char[16];
+            char[] name = new char[LENGTH.intValue()];
+            char[] mount = new char[LENGTH.intValue()];
+            WinNT.LARGE_INTEGER userFreeBytes = new WinNT.LARGE_INTEGER(0L);
+            WinNT.LARGE_INTEGER totalBytes = new WinNT.LARGE_INTEGER(0L);
+            WinNT.LARGE_INTEGER systemFreeBytes = new WinNT.LARGE_INTEGER(0L);
+        
+            Kernel32.INSTANCE.GetVolumeInformation(new String(volume).trim(), name, 255, null, null, null, fstype, 16);

```

Comment:
You keep using `new String(volume).trim()`.  How about making that a variable and passing the variable to all these methods instead of repeatedly making new Strings?  (Or, I guess the JVM will reuse the old string, but still, that's extra overhead!)